src/binary_tree/sparse_binary_tree.rs

```diff
@@ -0,0 +1,515 @@
+    Right,
+}
+
+impl<C: Clone> Node<C> {
+    /// Return true if self is a) a left sibling and b) lives just to the left of the other node.
+    #[allow(dead_code)]
+    pub fn is_left_sibling_of(&self, other: &Node<C>) -> bool {
+        match self.node_orientation() {
+            NodeOrientation::Left => {
+                self.coord.y == other.coord.y && self.coord.x + 1 == other.coord.x
+            }
+            NodeOrientation::Right => false,
+        }
+    }
+
+    /// Return true if self is a) a right sibling and b) lives just to the right of the other node.
+    pub fn is_right_sibling_of(&self, other: &Node<C>) -> bool {
+        match self.node_orientation() {
+            NodeOrientation::Left => false,
+            NodeOrientation::Right => {
+                self.coord.x > 0
+                    && self.coord.y == other.coord.y
+                    && self.coord.x - 1 == other.coord.x
+            }
+        }
+    }
+
+    /// Return the coordinates of this node's sibling, whether that be a right or a left sibling.
+    pub fn get_sibling_coord(&self) -> Coordinate {
+        match self.node_orientation() {
+            NodeOrientation::Left => Coordinate {
+                y: self.coord.y,
+                x: self.coord.x + 1,
+            },
+            NodeOrientation::Right => Coordinate {
+                y: self.coord.y,
+                x: self.coord.x - 1,
+            },
+        }
+    }
+
+    /// Return the coordinates of this node's parent.
+    /// The x-coord divide-by-2 works for both left _and_ right siblings because of truncation.
+    /// Note that this function can be misused if tree height is not used to bound the y-coord from above.
+    #[allow(dead_code)]
+    pub fn get_parent_coord(&self) -> Coordinate {
+        Coordinate {
+            y: self.coord.y + 1,
+            x: self.coord.x / 2,
+        }
+    }
+}
+
+impl<C: Clone> InputLeafNode<C> {
+    /// Convert the simpler node type to the actual Node type.
+    fn to_node(self) -> Node<C> {
+        Node {
+            content: self.content,
+            coord: Coordinate {
+                x: self.x_coord,
+                y: 0,
+            },
+        }
+    }
+}
+
+/// Used to orient nodes inside a sibling pair so that the compiler can guarantee a left node is actually a left node.
+pub enum Sibling<C: Clone> {
+    Left(LeftSibling<C>),
+    Right(RightSibling<C>),
+}
+
+/// Simply holds a Node under the designated 'LeftSibling' name.
+pub struct LeftSibling<C: Clone>(Node<C>);
+
+/// Simply holds a Node under the designated 'RightSibling' name.
+pub struct RightSibling<C: Clone>(Node<C>);
+
+/// A pair of sibling nodes, but one might be absent.
+pub struct MaybeUnmatchedPair<C: Mergeable + Clone> {
+    left: Option<LeftSibling<C>>,
+    right: Option<RightSibling<C>>,
+}
+
+/// A pair of sibling nodes where both are present.
+pub struct MatchedPair<C: Mergeable + Clone> {
+    left: LeftSibling<C>,
+    right: RightSibling<C>,
+}
+
```

Comment:
Add this logic to the tree builder file (may have to change things around a bit)